libs/content/workspace/src/workspace.rs

```diff
@@ -113,18 +121,52 @@ impl WsConfig {
         self.auto_sync.store(auto_sync, Ordering::Relaxed);
         self.zen_mode.store(zen_mode, Ordering::Relaxed);
     }
+
+    pub fn update_last_open_tabs(&mut self, tabs: &[Tab], active_tab_index: usize) {
+        let mut active_tab = None;
+        let mut tab_ids: Vec<Uuid> = tabs
+            .iter()
+            .enumerate()
+            .filter_map(|(i, t)| {
+                if i == active_tab_index {
+                    active_tab = Some(t.id);
+                    None
+                } else {
+                    Some(t.id)
+                }
+            })
+            .collect();
+
+        if let Some(tab) = active_tab {
+            tab_ids.push(tab);
+        }
+
+        self.last_open_tabs = Arc::new(tab_ids);
+
+        if let Err(msg) = self.to_file() {
+            println!("{:#?}", msg);
+        }
+    }
+
+    pub fn to_file(&self) -> io::Result<()> {
+        let content = serde_json::to_string(&self).ok().unwrap();
+        fs::write(&self.path, content)
+    }
 }
 
 impl Workspace {
-    pub fn new(cfg: WsConfig, core: &Lb, ctx: &Context) -> Self {
+    pub fn new(core: &Lb, ctx: &Context) -> Self {
         let (updates_tx, updates_rx) = channel();
         let background = BackgroundWorker::new(ctx, &updates_tx);
         let background_tx = background.spawn_worker();
         let status = Default::default();
         let output = Default::default();
 
+        let writable_dir = core.get_config().writeable_path;
+        let writeable_dir = Path::new(&writable_dir);
+        let writeable_path = writeable_dir.join("ws_conf.json");
```

Comment:
I think there's a better path forward here. Here are some of my problems with stuff I see:

- skip_serialization is a data modeling smell
- path containing `""` as an initial value is a data modeling smell (that I introduced)
- similar fields getting copied into a few places with different variations seems risky. ws_conf.json is defined here.

So here is what I would suggest:
- move settings into it's own file
- keep the data getting written to it's own file in it's own, clean struct, without any Atomic business, document the fields figure out a good name for this thing
- wrap this thing in what hangs on to  it's `Path`, hangs on to the thing getting written in an `Arc<RwLock<Thing>>`.
- I picture an API that let's you read a one-off setting, ergonomic functions for mutating the settings that kickoff the file update. These functions should take care of the lock for me, and should be named in a way that's easy to discover.
- Writes to the file should happen in a background thread.

Look at all the things we want to do with the idea of [Workspace Settings](https://github.com/lockbook/lockbook/issues/2936) we probably need a version field and that clean struct should probably have a `*V1` with it's name. We'll need to think about migrating this file as we evolve it to be more sophisticated.

Reply:
Agree there are some smells including existing smells. This is a good time to do cleanup because we're introducing persistence and because we'll probably start persisting lots of things (so investments here will pay off quickly).

+1 for JSON so we don't have to do the versions and migrations thing (down to debate).

Hesitant to call this 'config' or 'settings' because lots of persistent state (what tabs are open, where they're scrolled to) isn't exactly one of those things. For settings, I have similar API desires - I imagine writing a button to change a single setting and would like that button-writing experience to be smooth. For other persistent things, I like the "just save everything" direction. Possibly worthwhile to split persistent state along those lines and have the API style that makes the most sense for each part.

Reply:
+1 for JSON so we don't have to do the versions and migrations thing (down to debate).

good point I'm used to everything persistent not being like this